### PR TITLE
CLI Tool: Better error handling

### PR DESCRIPTION
**Pull request by jaceks2106**
Example
=
image

### CODE REVIEW

File: src/Kmd.Logic.Gateway.Automation/Models/ApiValidationResult.cs [hunk 1]
```diff
@@ -19,15 +19,18 @@ public class ApiValidationResult : ItemValidationResultBase
...
         public override string ToString()
         {
-            var api = $"API: '{this.Name}' - {this.Path}/{this.Version}";
-            var result = this.ToString(api);
-
-            result += this.ApiId.HasValue ? $"* Api ID: {this.ApiId.Value}" : string.Empty;
-            result += this.ApiVersionSetId.HasValue ? $"* API version set: {this.ApiVersionSetId.Value}" : string.Empty;
+            var result = $"* API Name: {this.Name}\n";
+            result += $"* Path: {this.Path}\n";
+            result += $"* Version: {this.Version}\n";
```
Comment (Araballi) [inline src/Kmd.Logic.Gateway.Automation/Models/ApiValidationResult.cs:24]: Use string builder, all places where we have more string concatenation

Comment (jaceks2106) [reply src/Kmd.Logic.Gateway.Automation/Models/ApiValidationResult.cs:24]: fixed

File: src/Kmd.Logic.Gateway.Automation/Models/ApiValidationResult.cs [hunk 2]
```diff
@@ -19,18 +20,21 @@ public class ApiValidationResult : ItemValidationResultBase
...
+            sb.AppendLine($"* Path: {this.Path}");
+            sb.AppendLine($"* Version: {this.Version}");
+            sb.Append(this.ApiId.HasValue ? $"* Api ID: {this.ApiId.Value}\n" : string.Empty);
+            sb.Append(this.ApiVersionSetId.HasValue ? $"* API version set: {this.ApiVersionSetId.Value}\n" : string.Empty);
+            sb.Append(base.ToString());
 
             foreach (var rev in this.Revisions)
             {
-                result += rev.ToString(api + ", ");
+                sb.AppendLine();
```
Comment (Araballi) [inline src/Kmd.Logic.Gateway.Automation/Models/ApiValidationResult.cs:33]: Info: Previous code don't have line , new line included here

File: src/Kmd.Logic.Gateway.Automation/Models/ProductValidationResult.cs [hunk 3]
```diff
@@ -10,12 +11,11 @@ public class ProductValidationResult : ItemValidationResultBase
...
         public override string ToString()
         {
-            var product = $"Product: {this.Name}";
-            var result = this.ToString(product);
-
-            result += this.ProductId.HasValue ? $"* Product ID: {this.ProductId.Value}\n" : string.Empty;
-
-            return result;
+            var sb = new StringBuilder();
+            sb.AppendLine($"* Product Name: {this.Name}");
```
Comment (Araballi) [inline src/Kmd.Logic.Gateway.Automation/Models/ProductValidationResult.cs:15]: Info: Previous code don't have line , new line included here

File: src/Kmd.Logic.Gateway.Automation/Models/ValidatePublishingResponse.cs [hunk 4]
```diff
@@ -15,31 +16,28 @@ public class ValidatePublishingResponse
...
                 }
-
-                result += "\n";
             }
 
             foreach (var api in this.Apis)
             {
-                result += api.ToString();
-                result += "\n";
+                sb.AppendLine(api.ToString());
```
Comment (Araballi) [inline src/Kmd.Logic.Gateway.Automation/Models/ValidatePublishingResponse.cs:30]: Info: Previous code don't have line , new line included here

File: src/Kmd.Logic.Gateway.Automation/Models/ValidatePublishingResponse.cs [hunk 5]
```diff
@@ -15,31 +16,28 @@ public class ValidatePublishingResponse
...
+                sb.AppendLine();
             }
 
             foreach (var product in this.Products)
             {
-                result += product.ToString();
-                result += "\n";
+                sb.AppendLine(product.ToString());
```
Comment (Araballi) [inline src/Kmd.Logic.Gateway.Automation/Models/ValidatePublishingResponse.cs:36]: Info: Previous code don't have line , new line included here

**Review by Araballi — APPROVED** [review]
LGTM

File: src/Kmd.Logic.Gateway.Automation/Models/ApiValidationResult.cs (hunk 2 above)
Comment (jaceks2106) [reply src/Kmd.Logic.Gateway.Automation/Models/ApiValidationResult.cs:33]: On purpose.

File: src/Kmd.Logic.Gateway.Automation/Models/ProductValidationResult.cs (hunk 3 above)
Comment (jaceks2106) [reply src/Kmd.Logic.Gateway.Automation/Models/ProductValidationResult.cs:15]: On purpose.

File: src/Kmd.Logic.Gateway.Automation/Models/ValidatePublishingResponse.cs (hunk 5 above)
Comment (jaceks2106) [reply src/Kmd.Logic.Gateway.Automation/Models/ValidatePublishingResponse.cs:36]: On purpose.

**Pull request merged**